tests.py

```diff
@@ -1446,14 +1458,26 @@ def test_main_run_with_token(monkeypatch, mock_requests_get, capsys):
     )
 
 
+def test_main_run_with_mismatched_token(monkeypatch, mock_requests_get,
+                                        capsys):
+    monkeypatch.setattr(sys, 'argv', [
+        'ghcloneall', '--user', 'test_user', '--concurrency=1',
+        '--github-token', 'fake-token',
+    ])
+    mock_requests_get.set_user('some-other-user')
```

Comment:
You know, having this test here I think we can avoid writing smart mock response classes by defaulting to .set_user(None) in the fixture, and setting it to something only in tests that actually provide a token.

(I apologize for the spam of multiple single comments, I don't know why I started doing that instead of a proper preview.  Too much GitLab at work?)

Reply:
Done